OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -119,6 +125,9 @@ public IEnumerable<IRenderable> Render(WorldRenderer wr)
 				}
 			}
 
+			if (Shrouded)
+				return NoRenderables;
```

Comment:
Unless I'm going crazy, the only method that calls this is `FrozenActorLayer`, which is where I think the check should be moved to.

Where does `FrozenUnderFog` call this?